formula-android/src/main/java/com/instacart/formula/integration/FormulaFragmentViewFactory.kt

```diff
@@ -0,0 +1,34 @@
+package com.instacart.formula.integration
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.instacart.formula.android.FeatureView
+import com.instacart.formula.android.ViewFactory
+import com.instacart.formula.fragment.FragmentKey
+import java.lang.IllegalStateException
+
+internal class FormulaFragmentViewFactory(
+    private val key: FragmentKey,
+    private val featureProvider: FeatureProvider,
+) : ViewFactory<Any> {
+
+    private var factory: ViewFactory<Any>? = null
+
+    @Suppress("UNCHECKED_CAST")
+    override fun create(inflater: LayoutInflater, container: ViewGroup?): FeatureView<Any> {
+        val featureEvent = featureProvider.getFeature(key) ?: throw IllegalStateException("Missing feature for unknown reasons.")
+        val viewFactory = factory ?: when (featureEvent) {
+            is FeatureEvent.MissingBinding -> {
+                throw IllegalStateException("Missing feature factory or integration for $key. Please check your FragmentFlowStore configuration.")
+            }
+            is FeatureEvent.Failure -> {
+                throw IllegalStateException("Feature failed to initialize: $key", featureEvent.error)
+            }
+            is FeatureEvent.Init -> {
+                featureEvent.feature.viewFactory as ViewFactory<Any>
+            }
+        }
+        this.factory = viewFactory
```

Comment:
nit: you can probably omit the `this` here, unless you're keeping it for readability reasons.

Reply:
Just to be explicit here, I ran into bugs where I would accidentally assign a local variable instead of the global one.